qabelbox/src/main/java/de/qabel/qabelbox/persistence/RepositoryFactory.java

```diff
@@ -0,0 +1,133 @@
+        try {
+            Class.forName("org.sqldroid.SQLDroidDriver");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    File getDatabasePath() {
+        return context.getFileStreamPath(DB_REPOSITORIES);
+    }
+
+    public void deleteDatabase() {
+        close();
+        try {
+            FileOutputStream outputStream = context.openFileOutput(DB_REPOSITORIES, Context.MODE_PRIVATE);
+            outputStream.write(1);
+            outputStream.close();
+        } catch (java.io.IOException ignored) {
+        }
+        if (!context.deleteFile(DB_REPOSITORIES)) {
+            throw new IllegalStateException("Could not delete client database");
+        }
+    }
+
+
+    public IdentityRepository getIdentityRepository(AndroidClientDatabase clientDatabase) {
+		SqliteDropUrlRepository dropUrlRepository = getSqliteDropUrlRepository(clientDatabase);
+		SqlitePrefixRepository prefixRepository = getSqlitePrefixRepository(clientDatabase);
+		IdentityHydrator hydrator = getIdentityHydrator(getEntityManager(), dropUrlRepository, prefixRepository);
+		return new SqliteIdentityRepository(
+				clientDatabase, hydrator, dropUrlRepository, prefixRepository);
+    }
+
+    @NonNull
+    public EntityManager getEntityManager() {
+        if (entityManager == null) {
+            entityManager = new EntityManager();
+        }
+        return entityManager;
+    }
+
+    @NonNull
+    private IdentityHydrator getIdentityHydrator(EntityManager em,
+                                                 SqliteDropUrlRepository dropUrlRepository,
+                                                 SqlitePrefixRepository prefixRepository) {
+        return new IdentityHydrator(
+                new DefaultIdentityFactory(),
+                em,
+                dropUrlRepository,
+                prefixRepository
+        );
+    }
+
+    @NonNull
+    public SqliteContactRepository getContactRepository(AndroidClientDatabase clientDatabase) {
+        return new SqliteContactRepository(clientDatabase, getEntityManager());
+    }
+
+    @NonNull
+    public SqlitePrefixRepository getSqlitePrefixRepository(AndroidClientDatabase clientDatabase) {
+        return new SqlitePrefixRepository(clientDatabase);
+    }
+
+    @NonNull
+    public SqliteDropUrlRepository getSqliteDropUrlRepository(AndroidClientDatabase clientDatabase) {
+        return new SqliteDropUrlRepository(clientDatabase, new DropURLHydrator());
+    }
+
+    @NonNull
+    public AndroidClientDatabase getAndroidClientDatabase() throws SQLException {
+        if (connection == null) {
+            connection = DriverManager.getConnection("jdbc:sqlite:" + getDatabasePath());
+            androidClientDatabase = new AndroidClientDatabase(connection);
+            try {
+                androidClientDatabase.migrate();
+            } catch (MigrationException e) {
+                throw new RuntimeException(e);
```

Comment:
This case leaves a connection and androidClientDatabase unmigrated. The next call to getAndroidClientDatabase() would return an androidClientDatabase without exception that is probably useless (because it is probably not fully migrated). I would prefer dropping the androidClientDatabase and retry migration on the next getAndroidClientDatabase() call. That would either work or throw an exception but not return an unmigrated instance